src/components/Common/SearchBarDropdown.tsx

```diff
@@ -0,0 +1,207 @@
+
+type SearchBarDropDownProps = {
+  placeholder: string;
+};
+
+const styles = {
+  autocomplete: {
+    "&.MuiAutocomplete-hasClearIcon .MuiOutlinedInput-root": { paddingX: 1 },
+    position: "relative",
+    padding: 0,
+  },
+  box: {
+    paddingBottom: 1,
+    paddingTop: 0,
+    paddingX: 2,
+  },
+  listBox: {
+    maxHeight: "60vh",
+  },
+  searchIcon: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "gray.main",
+    },
+  },
+  searchIconFocused: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "primary.main",
+    },
+  },
+};
+
+const SearchBarDropDown = ({ placeholder }: SearchBarDropDownProps) => {
+  const [focusSearchBar, setFocusSearchBar] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [keywords, setKeywords] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [searchOptions, setSearchOptions] = useState<SelectedUser[]>([]);
+  const { getAccessTokenSilently } = useAuth0();
+  const navigate = useNavigate();
+  const debouncedFetch = useMemo(
+    () => debounce((keywords: string) => fetchUsers(keywords), 100),
+    []
+  );
+
+  const fetchUsers = async (keywords: string) => {
+    setLoading(true);
+    if (keywords.length === 0) {
+      return;
+    }
+
+    try {
+      const token = await getAccessTokenSilently();
+      const result = await axios.get(
+        `http://localhost:3001/api/users/searchUsers`,
+        {
+          params: {
+            keyword: keywords,
+          },
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+
+      setSearchOptions(result.data as SelectedUser[]);
+
+      if (keywords.trim() === "") {
+        setSearchOptions([]);
+      }
+    } catch (error) {
+      console.error("Failed to fetch users:", error);
+      setSearchOptions([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    debouncedFetch(keywords);
+  }, [keywords]);
+
+  const handleInputChange = (newInputValue: string) => {
+    setKeywords(newInputValue);
+
+    if (newInputValue.trim() === "") {
+      setSearchOptions([]);
+    }
+  };
+
+  const handleSelect = (selectedUsername: string) => {
+    setKeywords("");
+    setSearchOptions([]);
+    inputRef.current?.blur();
+    navigate(`/${selectedUsername}`);
+  };
+
+  const handleClear = () => {
+    setKeywords("");
+    setSearchOptions([]);
+    setFocusSearchBar(false);
+  };
+
+  return (
+    <Box sx={styles.box}>
+      <Autocomplete
+        disablePortal
+        fullWidth
+        filterOptions={(x) => x}
+        getOptionLabel={() => ""}
+        forcePopupIcon={false}
+        id="search"
+        value={null}
+        inputValue={keywords}
+        ListboxProps={{ style: styles.listBox }}
+        loading={!keywords && loading}
+        loadingText="Start typing to search..."
+        onChange={(_, value) => value && handleSelect(value.username)}
+        onBlur={handleClear}
```

Comment:
Do we want the input field to clear on blur? Do you think it'd be better if we kept the input field and the options as is and close the dropdown on blur instead? I don't really have a strong opinion on this, so I'd be happy with either one. Let me know your thoughts!

Reply:
I guess from a user standpoint it would be kinda helpful if you are half-way through a search and you want to quickly look at something else